### PR TITLE
recommend RSA-PSS over PKCS#1

### PR DESCRIPTION
**Pull request by thomas-fossati**
An attempt at addressing #249

### CODE REVIEW

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 1]
```diff
@@ -349,6 +349,44 @@ The following recommendations are provided to help prevent SSL Stripping (an att
...
+alternative to RSASSA-PKCS1-v1_5.
+
+If supported by its PKI and/or TLS stacks, a TLS 1.2 Endpoint SHOULD offer
+and/or accept an RSASSA-PSS algorithm in both signature_algorithms and
+signature_algorithms_cert extensions.  When offering, the TLS 1.2 client
+SHOULD prioritise RSASSA-PSS over RSASSA-PKCS1-v1_5 algorithms (i.e., by
+setting the values at a lower index in the list).
+
+Rationale: RSA-PSS provides a provably secure alternative to PKCS#1 v1.5 and,
+when available, it be used instead.  TLS 1.3 made RSA-PSS
```
Comment (stpeter) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:387]:
```suggestion
when available, it should be used instead.  TLS 1.3 made RSA-PSS
```

**Comment by yaronf** [comment]
Guys, please hold this PR until I'm back.
Get Outlook for Android

**Comment by thomas-fossati** [comment]
> Guys, please hold this PR until I'm back. Get Outlook for Android

Sure, I just threw together a few thoughts to try and articulate Martin's point and have something concrete to discuss.

**Comment by stpeter** [comment]
Actually I think we'll hold all PRs until Yaron is back, but it can't hurt to propose text between now and then. :-)

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 2]
```diff
@@ -349,6 +349,44 @@ The following recommendations are provided to help prevent SSL Stripping (an att
...
 
+## Signature Algorithms
+
+{{Section 4.2.3 of RFC8446}} revamps the signature_algorithms
+extension first defined in {{Section 7.4.1.4.1 of RFC5246}} by
+merging the previously separate SignatureAlgorithm and HashAlgorithm
+into a single (backwards compatible) code point named
+SignatureScheme.  It also introduces the new
+signature_algorithms_cert extension that has the same syntax as
+signature_algorithms but slightly different semantics.  In fact,
```
Comment (thomas-fossati) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:360]:
```suggestion
signature_algorithms but slightly different semantics.  Specifically,
```

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 3]
```diff
@@ -349,6 +349,44 @@ The following recommendations are provided to help prevent SSL Stripping (an att
...
+
+If supported by its PKI and/or TLS stacks, a TLS 1.2 Endpoint SHOULD offer
+and/or accept an RSASSA-PSS algorithm in both signature_algorithms and
+signature_algorithms_cert extensions.  When offering, the TLS 1.2 client
+SHOULD prioritise RSASSA-PSS over RSASSA-PKCS1-v1_5 algorithms (i.e., by
+setting the values at a lower index in the list).
+
+Rationale: RSA-PSS provides a provably secure alternative to PKCS#1 v1.5 and,
+when available, it should be used instead.  TLS 1.3 made RSA-PSS
+mandatory-to-implement and provided the means to use it with TLS 1.2 also.
```
Comment (thomas-fossati) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:388]:
```suggestion
when available, it should be used instead.  TLS 1.3 makes RSA-PSS
mandatory-to-implement and provides the means to use it with TLS 1.2 also.
```

**Pull request closed without merging**